lib/gadget/gadget.vala

```diff
@@ -398,8 +398,12 @@ namespace Frida.Gadget {
 		}
 
 		public async void start () throws Error {
+			string listen_address = GLib.Environment.get_variable ("FRIDA_GADGET_LISTEN_ADDRESS");
+			if (listen_address == null){
```

Comment:
I agree that'd be better.

Would you prefer the parsing done only before starting the server, i.e. after https://github.com/frida/frida-core/blob/master/lib/gadget/gadget.vala#L102 , or parse it anyway?